core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -266,98 +246,234 @@ func (i *pluginOracleCreator) createReadersAndWriters(
+}
+
+func (i *pluginOracleCreator) getChainReaderConfig(
+	chainID uint64,
+	destChainID uint64,
+	ofc offChainConfig,
+	chainSelector cciptypes.ChainSelector,
+) evmrelaytypes.ChainReaderConfig {
+	var chainReaderConfig evmrelaytypes.ChainReaderConfig
+	if chainID == destChainID {
+		chainReaderConfig = evmconfig.DestReaderConfig
+	} else {
+		chainReaderConfig = evmconfig.SourceReaderConfig
+	}
+
+	if !ofc.commitEmpty() && ofc.commit().PriceFeedChainSelector == chainSelector {
+		chainReaderConfig = evmconfig.MergeReaderConfigs(chainReaderConfig, evmconfig.FeedReaderConfig)
+	}
+	return chainReaderConfig
+}
+
+func (i *pluginOracleCreator) createChainReader(
+	chain legacyevm.Chain,
+	chainReaderConfig evmrelaytypes.ChainReaderConfig,
+	pluginType cctypes.PluginType,
+) (types.ContractReader, error) {
+	cr, err := evm.NewChainReaderService(
+		context.Background(),
+		i.lggr.
+			Named("EVMChainReaderService").
+			Named(chain.ID().String()).
+			Named(pluginType.String()),
+		chain.LogPoller(),
+		chain.HeadTracker(),
+		chain.Client(),
+		chainReaderConfig,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create contract reader for chain %s: %w", chain.ID(), err)
+	}
+	return cr, nil
+}
+
+func (i *pluginOracleCreator) bindContracts(
+	chain legacyevm.Chain,
+	cr types.ContractReader,
+	config cctypes.OCR3ConfigWithMeta,
+	destChainID uint64,
+	ofc offChainConfig,
+	chainSelector cciptypes.ChainSelector,
+) error {
+	if chain.ID().Uint64() == destChainID {
+		offrampAddressHex := common.BytesToAddress(config.Config.OfframpAddress).Hex()
+		err := cr.Bind(context.Background(), []types.BoundContract{
+			{
+				Address: offrampAddressHex,
+				Name:    consts.ContractNameOffRamp,
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to bind chain reader for dest chain %s's offramp at %s: %w", chain.ID(), offrampAddressHex, err)
+		}
+	}
 
-		chainSelector, ok := chainsel.EvmChainIdToChainSelector()[chain.ID().Uint64()]
-		if !ok {
-			return nil, nil, fmt.Errorf("failed to get chain selector from chain ID %s", chain.ID())
+	if !ofc.commitEmpty() && chainSelector == ofc.commit().PriceFeedChainSelector {
+		tokenInfo := ofc.commit().TokenInfo
+		var bcs []types.BoundContract
+		for _, info := range tokenInfo {
+			bcs = append(bcs, types.BoundContract{
+				Address: info.AggregatorAddress,
+				Name:    consts.ContractNamePriceAggregator,
+			})
+		}
+		if err := cr.Bind(context.Background(), bcs); err != nil {
+			return nil
 		}
+	}
+	return nil
+}
 
-		contractReaders[cciptypes.ChainSelector(chainSelector)] = cr
-		chainWriters[cciptypes.ChainSelector(chainSelector)] = cw
+func (i *pluginOracleCreator) createChainWriter(
+	chain legacyevm.Chain,
+	pluginType cctypes.PluginType,
+	execBatchGasLimit uint64,
+) (types.ChainWriter, error) {
+	var fromAddress common.Address
+	transmitter, ok := i.transmitters[types.NewRelayID(relay.NetworkEVM, chain.ID().String())]
+	if ok {
+		fromAddress = common.HexToAddress(transmitter[0])
```

Comment:
Not sure if there was a TODO or not but lets a TODO here to remove EVM-specific stuff.